src/scripts/faucet.ts

```diff
@@ -0,0 +1,156 @@
+import { createPublicClient, getContract, http } from "viem";
+import { diamondAddress, ufaucetAbi } from "./constants";
+import { mainnet } from "viem/chains";
+import { getConnectedClient } from "./connect-wallet";
+
+const abi = ufaucetAbi;
+const address = diamondAddress;
+const publicClient = createPublicClient({
+  chain: mainnet,
+  transport: http(),
+});
+
+const CLIENT_OR_ACCOUNT_ERROR = "Client or account not initialized";
+
+export async function getAllCollaterals() {
```

Comment:
You are creating a contract instance and a public function for every single contract method, there is no need to do that. [As in viem docs](https://viem.sh/docs/contract/getContract.html), you should make a public contract instance and then you can directly call it's methods. From their official documentation:

```
import { getContract } from 'viem'
import { wagmiAbi } from './abi'
import { publicClient, walletClient } from './client'
 
// 1. Create contract instance
const contract = getContract({
  address: '0xFBA3912Ca04dd458c843e2EE08967fC04f3579c2',
  abi: wagmiAbi,
  // 1a. Insert a single client
  client: publicClient,
  // 1b. Or public and/or wallet clients
  client: { public: publicClient, wallet: walletClient }
})
 
// 2. Call contract methods, fetch events, listen to events, etc.
const result = await contract.read.totalSupply()
const logs = await contract.getEvents.Transfer()
const unwatch = contract.watchEvent.Transfer(
  { from: '0xA0Cf798816D4b9b9866b5330EEa46a18382f251e' },
  { onLogs(logs) { console.log(logs) } }
)
```